sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -309,6 +309,15 @@ static struct snd_soc_dai_link *sof_card_dai_links_create(struct device *dev,
 	links[id].dpcm_playback = 1;
 	links[id].dpcm_capture = 1;
 	links[id].no_pcm = 1;
+	/*
+	 * Currently, MCLK will be turned off in sof runtime suspended, and
+	 * it will go into runtime suspended right after playback is stop.
+	 * However, rt5682 will output static noise if sysclk turns off during
+	 * playback. Set ignore_pmdown_time to power down rt5682 immediately
+	 * and avoid the noise.
+	 * It can be removed once we can control MCLK by driver.
+	 */
+	links[id].ignore_pmdown_time = 1;
```

Comment:
this should only be set for SKL+, for legacy platforms this is not needed, and actually you can request the MCLK on demand, see e.g. bytcr_rt5640

Also can you rework the code to use the codec components instead of codec_name and codec dai_name.